### PR TITLE
Added eslint-plugin-react and eslint-plugin-react-hooks.

### PR DESCRIPTION
**Pull request by nicolad**
## Checklist

* [ ] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [ ] My code follows the Event Espresso code style.
* [ ] My code has proper inline documentation.
* [ ] My code accounts for when the site is in Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

**Comment by nicolad** [comment]
I turned `off` `comma-dangle` rule otherwise we've got to deal with a lot of errors:

image

**Comment by nicolad** [comment]
There are a number of warnings.
We can disable some of which we might think are not relevant, but also add new ones, like `no-magic-numbers`. That needs to be done in a separate PR otherwise, this would get messy and hard to follow.

image

**Comment by nicolad** [comment]
This PR needs to be merged after this one will get merged: https://github.com/eventespresso/event-espresso-core/pull/1840

**Comment by nicolad** [comment]
@tn3rb @manzoorwanijk I need your blessings here.

**Comment by tn3rb** [comment]
> This PR needs to be merged after this one will get merged: #1840

can't do it now because that PR has already been merged, but in the future you can set another PR as a dependency and it will prevent your PR from getting merged prior to the other PR
